pkg/controller/database/rdsinstance.go

```diff
@@ -211,8 +213,23 @@ func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 	if err != nil {
 		return managed.ExternalUpdate{}, errors.Wrap(err, errPatchCreationFailed)
 	}
-	modify := e.client.ModifyDBInstanceRequest(rds.GenerateModifyDBInstanceInput(meta.GetExternalName(cr), patch))
-	_, err = modify.Send(ctx)
+	_, err = e.client.ModifyDBInstanceRequest(rds.GenerateModifyDBInstanceInput(meta.GetExternalName(cr), patch)).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errModifyFailed)
+	}
+	if len(patch.Tags) > 0 {
```

Comment:
Nit: This conditional block is almost half of the method. Should we return early if there are zero tags instead of nesting all of this logic?